### PR TITLE
Implement file upload

### LINKED ISSUE #24: Implement `SSH_FXP_WRITE` handler
Supporting this will allow SFTP clients to write data to Permanent records.

### LINKED ISSUE #48: Implement file open / write handlers
This will allow users to create records on their permanent accounts.

### PR DESCRIPTION
**Pull request by slifty**
This PR adds functionality for writing files to Permanent.

It has a few known limitations (which have been captured in issues already):

1. It does not support empty file uploads (see #79)
2. It does not support `mod time` (see #80)
3. It does not support changes to existing files in permanent (see https://github.com/PermanentOrg/sftp-service/issues/81)

Resolves #48 
Resolves #24

### CODE REVIEW

File: src/classes/PermanentFileSystem.ts [hunk 1]
```diff
@@ -103,7 +105,7 @@ export class PermanentFileSystem {
     if (targetIsFolder) {
       return fs.constants.S_IFDIR;
     }
-    throw new Error('Item was not found');
+    return undefined;
```
[the remark this answers is not in the document]
Comment (slifty) [reply src/classes/PermanentFileSystem.ts:108]: The thinking was that if the item doesn't exist, its type would not be defined.

That said, I suppose the originally thrown error would be more expected behavior.  I can revert to that!

[the remark this answers is not in the document]
Comment (slifty) [reply src/classes/PermanentFileSystem.ts]: Fixed up!

File: src/classes/SftpSessionHandler.ts [hunk 2]
```diff
@@ -481,4 +609,140 @@ export class SftpSessionHandler {
...
+            );
+            this.sftpConnection.handle(
+              reqId,
+              Buffer.from(handle),
+            );
+            break;
+          // We do not currently allow anybody to edit an existing record in any way
+          case 'r+': // read and write
+          case 'w': // write
+          case 'w+': // write and red
```
Comment (Fenn-CS) [inline src/classes/SftpSessionHandler.ts:639]: Small typo in comment

**Review by Fenn-CS — APPROVED** [review]
Code looks good, left two inline comments one of which you already resolved.

Tested by uploading a file to prod! Hurray!


Thanks @slifty

**Pull request merged**